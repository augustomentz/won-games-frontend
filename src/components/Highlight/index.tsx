import * as S from './styles'

import Button from 'components/Button'

export type HighlightProps = {
	title: string
	subtitle: string
	bgImage: string
	floatImage?: string
	buttonLabel: string
	buttonLink: string
	alignment?: 'right' | 'left'
}

const Highlight = ({
	title,
	subtitle,
	bgImage,
	floatImage,
	buttonLabel,
	buttonLink,
	alignment = 'right'
}: HighlightProps) => (
	<S.Wrapper bgImage={bgImage} alignment={alignment}>
		{!!floatImage && <S.FloatImage src={floatImage} alt={title} />}
		<S.Content>
			<S.Title>{title}</S.Title>
			<S.Subtitle>{subtitle}</S.Subtitle>
			<Button as="a" href={buttonLink}>
				{buttonLabel}
			</Button>
		</S.Content>
	</S.Wrapper>
)

export default Highlight
