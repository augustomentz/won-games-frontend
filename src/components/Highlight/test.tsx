import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Highlight from '.'

const props = {
	title: 'Heading 1',
	subtitle: 'Heading 2'
}

describe('<Highlight />', () => {
	it('Should be render the heading', () => {
		const { container } = renderWithTheme(<Highlight {...props} />)

		expect(
			screen.getByRole('heading', { name: /Heading 1/i })
		).toBeInTheDocument()
		expect(
			screen.getByRole('heading', { name: /Heading 2/i })
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
