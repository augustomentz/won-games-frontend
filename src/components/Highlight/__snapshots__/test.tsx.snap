// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> Should be render the heading 1`] = `
<main
  class=""
>
  <h1>
    Highlight
  </h1>
</main>
`;
